sympy/physics/vector/tests/test_functions.py

```diff
@@ -488,3 +490,15 @@ def test_partial_velocity():
 
     raises(TypeError, lambda: partial_velocity(Dmc.vel(N), u_list, N))
     raises(TypeError, lambda: partial_velocity(vel_list, u1, N))
+
+
+def test_different_time_symbol():
+    import sympy.physics.vector.functions
+    t = sympy.physics.vector.functions.TIME
+    x = Symbol('x')
+    sympy.physics.vector.functions.TIME = x
```

Comment:
We shouldn't alter global state like this in tests. More care is needed to ensure that this global state is restored.

Reply:
Care was taken with line 499. If you have suggestion for better care, I'm happy to implement it.

Reply:
This test should use try/finally to restore the state.

If we really want to support changing the time symbol then providing a context manager for doing so might be a good idea. That would give the option to add things like thread safety later...

Reply:
I'd be willing to explore a context manager. Thoughts on what that would look like?

Reply:
I guess it would look like:
```
@contextmanager
def set_time_symbol(sym):
    old_sym = functions.TIME
    try:
        functions.TIME = sym
        yield
    finally:
        functions.TIME = old_sym

with set_time_symbol(tau):
    ...
```
That means that a user doesn't need to directly modify the TIME variable. If you also provide a function like `get_time_symbol()` then users don't even need to know that it's a global variable in the functions module.

If the longer term plan though is to add a time parameter to the relevant APIs (a better ultimate solution IMO) then we shouldn't add any API for manipulating TIME now.